File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head.next == null){
            head= null;
            return head;
        }
        int size = 0;
        ListNode current = head;
        while(current != null){
            size++;
            current = current.next;
        }
        if(size == n){
            return head.next;
        }

        ListNode prev = null;
        current = head;
        for(int i = 0; i<size - n;i++){
            prev = current;
            current = current.next;
        }
        prev.next = current.next;
        return head;
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        cur1 = l1.next
        cur2 = l2.next

        total = l1.val + l2.val 
        carry = total // 10

        head2 = ListNode(total % 10)

        cur3 = head2

        while cur1 or cur2:

            val1 = cur1.val if cur1 else 0
            val2 = cur2.val if cur2 else 0

            total = val1 + val2 + carry
            carry = total // 10

            cur3.next = ListNode(total % 10)
            cur3 = cur3.next

            if cur1:
                cur1 = cur1.next
            if cur2 :
                cur2 =cur2.next

        if carry != 0:
            cur3.next = ListNode(carry)

        return head2


